File: python/square.py
"""
Name: Alexander Katrompas
Assignment: Commenting Demonstration
Purpose: Demonstrates a properly constructed and
         documented class module.
"""

DEFAULT_LENGTH = 1.0


class Square:
    """
    Represent a square with a configurable side length.

    The area is calculated on demand rather than stored.
    A non-positive construction value leaves the square
    at the default side length.

    :note: none
    """

    def __init__(self, length: float) -> None:
        """
        Initialize a square with the specified side length.

        :param length: the initial side length of the square
        :raises: none
        :return: None
        :note: a non-positive value leaves the length set to
               DEFAULT_LENGTH
        """
        self._length: float = DEFAULT_LENGTH
        self.set_length(length)

    def get_area(self) -> float:
        """
        Return the area of the square.

        :param: none
        :raises: none
        :return: the area of the square
        :note: the area is calculated on demand
        """
        return self._length * self._length

    def get_length(self) -> float:
        """
        Return the side length of the square.

        :param: none
        :raises: none
        :return: the side length of the square
        :note: none
        """
        return self._length

    def set_length(self, length: float) -> None:
        """
        Set the side length of the square.

        :param length: the new side length of the square
        :raises: none
        :return: None
        :note: a non-positive value leaves the current length
               unchanged
        """
        if length > 0:
            self._length = length

File: c++/square.cpp
/**
 * Name: Alexander Katrompas
 * Assignment: Demonstration Code
 * Purpose: Demonstrates a properly constructed and
 *          commented class definition file, Square.cpp.
 */

#include "square.h"

/**
 * Constructs a square with the specified side length.
 *
 * Non-positive values cause the length to be set to
 * the default value DLENGTH.
 *
 * @param length the length of a side of the square
 * @exception none
 * @return none
 * @note none
 */
Square::Square(float length) {
    setLength(length);
}

/**
 * Destroys the square.
 *
 * @param none
 * @exception none
 * @return none
 * @note none
 */
Square::~Square() {
}

/**
 * Returns the area of the square.
 *
 * The area is calculated on demand rather than stored
 * as an attribute.
 *
 * @param none
 * @exception none
 * @return the area of the square
 * @note none
 */
float Square::getArea() const {
    return length * length;
}

/**
 * Returns the side length of the square.
 *
 * @param none
 * @exception none
 * @return the side length of the square
 * @note none
 */
float Square::getLength() const {
    return length;
}

/**
 * Sets the side length of the square.
 *
 * Non-positive values cause the length to be set to
 * the default value DLENGTH.
 *
 * @param length the new side length of the square
 * @exception none
 * @return void
 * @note none
 */
void Square::setLength(float length) {
    if (length > 0) {
        this->length = length;
    } else {
        this->length = DLENGTH;
    }
}